plugins/node/opentelemetry-instrumentation-connect/test/instrumentation.test.ts

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as assert from 'assert';
+
+import { context, trace } from '@opentelemetry/api';
+import { RPCType, setRPCMetadata } from '@opentelemetry/core';
+import { SemanticAttributes } from '@opentelemetry/semantic-conventions';
+import { AsyncHooksContextManager } from '@opentelemetry/context-async-hooks';
+import { NodeTracerProvider } from '@opentelemetry/node';
+import {
+  InMemorySpanExporter,
+  SimpleSpanProcessor,
+} from '@opentelemetry/tracing';
+import * as http from 'http';
+import { createServer } from 'net';
+import { ANONYMOUS_NAME, ConnectInstrumentation } from '../src';
+
+const server = createServer();
+const httpRequest = {
+  get: (options: http.ClientRequestArgs | string) => {
+    return new Promise((resolve, reject) => {
+      return http.get(options, resp => {
+        let data = '';
+        resp.on('data', chunk => {
+          data += chunk;
+        });
+        resp.on('end', () => {
+          resolve(data);
+        });
+        resp.on('error', err => {
+          reject(err);
+        });
+      });
+    });
+  },
+};
+
+const getNextPort = (port: number, count = 0) => {
+  if (count > 10) {
+    return Promise.reject('too many retries');
+  }
+  return new Promise((resolve, reject) => {
+    const timer = setTimeout(() => {
+      getNextPort(++port, ++count).then(resolve, reject);
```

Comment:
You can ask for a random number by using 0 for the port number and then later reading the actual used port from the initiated server.

Reply:
it would be just additional logic, whereas it really doesn't matter, it is useful especially during testing when something hangs etc.

Reply:
You are searching for an open port which can fail. What @Rauno56 is saying is you can start the server with port 0 and it is guaranteed to be assigned to an open port.

Reply:
The current solution(using another server to find the port and then closing it) works as well I guess, but you could just start the server you will be using for requests on port 0 and detect the port it's running instead of allocating, starting and closing another server.

Reply:
I had already changed that 5 days ago guys .... :)

Reply:
I know - my last comment was about that changed version.